validator/error.go

```diff
@@ -0,0 +1,23 @@
+package validator
+
+import "errors"
+
+// Define errors
+var (
```

Comment:
[Style][ Errors can be local to the rules, for maintenance and redability.
This is only my own opinion

Reply:
So let's not define them and do hardcode?

Reply:
define them.

Folder structure:
rules
   exist
        rule.go
        errros.go
        utils etc

Reply:
I think we can do this. As utils have same struct for all the rules. WDYT?
```
validator
     validator.go
     rules
          length.go
          exist.go
          utils.go
```

Reply:
And also some errors defined will be used by all rules, like ErrFieldNotExist, ErrUnsupported, ErrAbsent etc. So ig it's better to have errors defined commonly?